packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -416,36 +416,129 @@ class DaemonDomain extends Domain {
+              canBeEnabledWithCreate = false;
+            }
+            if (!supportedPlatforms.contains(SupportedPlatform.android)) {
+              isSupported = false;
+              reasons.add('the current Flutter project does not have an Android platform directory');
+            }
+          case PlatformType.web:
+            if (!featureFlags.isWebEnabled) {
+              isSupported = false;
+              reasons.add('the Web feature is not enabled ("flutter config --enable-web")');
+              canBeEnabledWithCreate = false;
+            }
+            if (!supportedPlatforms.contains(SupportedPlatform.web)) {
+              isSupported = false;
+              reasons.add('the current Flutter project does not have a Web platform directory');
+            }
+          case PlatformType.fuchsia:
+            if (!featureFlags.isFuchsiaEnabled) {
+              isSupported = false;
+              reasons.add('the Fuchsia feature is not enabled ("flutter config --enable-fuchsia")');
+              canBeEnabledWithCreate = false;
+            }
+            if (!supportedPlatforms.contains(SupportedPlatform.fuchsia)) {
+              isSupported = false;
+              reasons.add('the current Flutter project does not have a Fuchsia platform directory');
+            }
+          case PlatformType.custom:
+            if (!featureFlags.areCustomDevicesEnabled) {
+              isSupported = false;
+              reasons.add('the custom-devices feature is not enabled ("flutter config --enable-custom-devices")');
+              canBeEnabledWithCreate = false;
+            }
+          case PlatformType.windowsPreview:
+            // TODO(fujino): detect if there any plugins with native code
+            if (!featureFlags.isPreviewDeviceEnabled) {
+              isSupported = false;
+              reasons.add('the flutter-preview feature is not enabled ("flutter config --enable-flutter-preview")');
+              canBeEnabledWithCreate = false;
+            }
+            if (!supportedPlatforms.contains(SupportedPlatform.windows)) {
+              isSupported = false;
+              reasons.add('the current Flutter project does not have a Windows platform directory');
+            }
+        }
+
+        if (isSupported) {
+          platforms.add(platform.name);
+          platformTypesMap[platform.name] = const <String, Object>{
+            'isSupported': true,
+          };
+        } else {
+          platformTypesMap[platform.name] = <String, Object>{
+            'isSupported': false,
+            'reason': reasons.join(' and '),
+            'canBeEnabledWithCreate': canBeEnabledWithCreate,
+          };
+        }
       }
+
+      PlatformType.values.forEach(handlePlatformType);
+
       return <String, Object>{
-        'platforms': result,
+        // TODO(fujino): delete the "platforms" entry once all daemon clients
+        // have adopted "platformTypes"
```

Comment:
Nit: If you indent the second line of a TODO with an extra space, the analysis server will know it's a continuation and "unwrap" the text in the Problems window (if enabled).

We also need to keep this around for some period to account for people on old versions of the plugins - we probably want > 12mo from when the plugins ship using this (FWIW, I highly suspect VS Code is the only user of this API).

Reply:
understood, i just wanted to put something in the TODO text. I'll instead just link to an issue. I will not delete this without verifying with you first :)

Reply:
updated